ambry-frontend/src/main/java/com/github/ambry/frontend/GetBlobHandler.java

```diff
@@ -243,6 +243,9 @@ private Callback<Void> securityProcessResponseCallback(GetBlobResult result) {
             ReadableStreamChannel response = result.getBlobDataChannel();
             if (subResource == null) {
               response = closeChannelIfNotModified(result);
+              if (response != null) {
+                restRequest.getMetricsTracker().setBytesTransferred(result.getBlobInfo().getBlobProperties().getBlobSize());
```

Comment:
BlobSize in blob properties might not be the right bytesTransferred when it's range request. Client can specify a range to download when issuing a get request, in this way, blob size in blob properties are still the length for entire blob. We have to find a different way to do this

And it seems like there is easy way to do this by looking at the ReadableStreamChannel.getSize, since this method would return -1 for range request as well. 

Hmm, might have to resolve it by using the NettyResponseChannel.totalBytesReceived anyway.